diff --git a/src/include/fst/types.h b/src/include/fst/types.h
index 9c0b799..d81dbf2 100644
--- a/src/include/fst/types.h
+++ b/src/include/fst/types.h
@@ -35,7 +35,8 @@ using uint64 = uint64_t;
 #ifdef _MSC_VER
 // Not really Windows-specific: they should have used ptrdiff_t in the first
 // place. But on Windows there has never been ssize_t.
-using ssize_t = std::ptrdiff_t;
+#include <basetsd.h>
+#define ssize_t SSIZE_T
 #endif  // _MSC_VER
 
 #endif  // FST_LIB_TYPES_H_
